Extensions/CompositeMetadata.md

```diff
@@ -0,0 +1,45 @@
+# Composite Metadata Extension
+
+_This extension specification is currently incubating._
+
+## Introduction
+There are a number of situations where an arbitrary collection of discrete metadata types should be attached to frame.  For example, a request frame may want to include both routing metadata as well as tracing metadata.  This extension specification provides an interoperable structure for metadadata payloads to contain multiple discrete metadata types.  It is designed such that if a consumer of the metadata is unaware of a particular type, it can be safely skipped and the next one read.
```

Comment:
Is there guidance on using this for single metadata blocks? MAY be used?

Reply:
If you're using a single metadata type, you would just write the metadata without using this type at all, right?  I'm not sure the spec needs to define, when _not_ to use it?

Reply:
OK, just raising.  I tend to find every potential ambiguity like this ends up with an implementor making different choices e.g. you receive this block with 0 metadata payloads instead of changing the top level flag :)

But TIOLI

Reply:
Given that we're still in the situation where the metadata MIME type for a connection is defined as part of `SETUP` I think we're going to have 0 and 1 inner metadatas when payloads are not homogenous.

@rdegnan, @robertroeser, @smaldini, or @stevegury any tie-breaking opinions?  Should we make some statement about using composite metadata in situations where it is unnecessary, but otherwise harmless (other than wasted bytes for a single)?

Reply:
One difference between this and setup is that setup comes one time. Heres a list of well know mime types:

http://hul.harvard.edu/ois/////systems/wax/wax-public-help/mimetypes.htm

In addition to these we can add our own and then base the know list of the version. 

I would be ok with purposing this for rsocket setup as well.

Reply:
@robertroeser I'm not sure I'm following you here.  Are you stating that you'd like to support the entire IETF MIME type registry (as a starting point) as well-known MIME types?  That any of them could have an integer value equivalent?  I'd assumed that most types would be done with strings but spec'd extensions could have a well-known id since they're close to the project.

Reply:
I liked the explicit mapping of types related to rsocket.  But suggest we extend it to support some well known ones likely to be used e.g. application/octet-stream, json, xml, etc + our zipkin format.

Reply:
@nebhale @yschimke  If we are going to support only certain types, then we never send a mime-type string and only send binary. If we are going to allow a string and arbitrary types then I suggest we create an integer for all the well know types in addition to the ones we support so we aren't send tons of extra data.

Reply:
I have some concern about trying to assign an integer to every well-known MIME types as that list is fluid.  It grows, so we'd have to update our integer assignments regularly to support that growth.  Not sure how we'd version our well known-to-integer mapping either.

Reply:
> I have some concern about trying to assign an integer to every well-known MIME types as that list is fluid.
By well know mime-types - I mean the commonly used ones - that's the list I included as a starting place, and come up with a list.

> Not sure how we'd version our well known-to-integer mapping either.
You create a static table and then you increment an integer as you add more types that you want to support. If its not on the table you send a string.